notebooks/svo-ranking/02_ranking_suggestions.py

```diff
@@ -3,39 +3,36 @@
 import os
 import pandas as pd
 
+DIR_RAW = os.getenv('DIR_DATA_RAW', 'data/raw')
+DIR_PROCESSED = os.getenv('DIR_DATA_PROCESSED', 'data/processed')
 
-DIR_RAW = os.getenv('DIR_DATA_RAW')
-DIR_PROCESSED = os.getenv('DIR_DATA_PROCESSED')
-
-# get GA and content store data
+# get GA data
 df_ga = pd.read_csv(filepath_or_buffer=DIR_RAW + '/bq_results_svo_ranking.csv')
-df_cs = pd.read_csv(filepath_or_buffer=DIR_RAW + '/preprocessed_content_store_010221.csv.gz',
-                    compression='gzip',
-                    encoding='utf-8',
-                    sep='\t',
-                    header=0,
-                    names=list(CONTENT_STORE_HEADERS.keys()),
-                    dtype=CONTENT_STORE_HEADERS,
-                    parse_dates=CONTENT_STORE_DATES)
+# Remove pages we don't want (search pages or smart answers etc)
+# Reduces numbers from ~90m rows to ~14m rows
+df_ga = df_ga[~df_ga.pageId.str.contains("\?", na=False)]
+df_ga = df_ga[~df_ga.pageId.str.contains("/y/", na=False)]
+df_ga = df_ga[~df_ga.pageId.str.contains("/n/", na=False)]
+df_ga = df_ga[~df_ga.pageId.str.contains("/[date]/", na=False)]
+df_ga = df_ga[~df_ga.pageId.str.contains("/email/unsubscribe/", na=False)]
+df_ga = df_ga[~df_ga.pageId.str.contains("/email-signup", na=False)]
 
-# join first_published_at
+# Add path
 df_ga['pagePath'] = df_ga['pageId'].str.split('- ').str[-1]
-df_ga = df_ga.merge(right=df_cs[['base_path', 'first_published_at', 'title']],
-                    how='left',
-                    left_on='pagePath',
-                    right_on='base_path',
-                    validate='many_to_one')
 
-df_ga = df_ga.rename(columns={'first_published_at': 'pageFirstPublishedAt',
-                              'title': 'pageTitle'})
-df_ga[['visitStartDateWeek',
-       'pageId',
-       'pagePath',
-       'pageTitle',
-       'pageFirstPublishedAt',
-       'pageHits',
-       'pageHitsMean',
-       'allPageHitsMin',
-       'allPageHitsMax',
-       'pageHitsNormalised']].to_csv(path_or_buf=DIR_PROCESSED + '/svo_ranking.csv',
-                                     index=False)
+# group and find the mean normalised hits across all the weeks for the time period
+df_ga['averagePageHitsMeanNormalised'] = df_ga.groupby('pagePath')['pageHitsMeanNormalised'].transform('mean')
```

Comment:
Why do we care what the page views where a year ago? Or even a two weeks ago? Is this Avision practicing for his forecasting job at Lego here?! haha

We could do this all in bigquery in a oner too